crates/sui/src/keytool.rs

```diff
@@ -541,41 +555,45 @@ impl KeyToolCommand {
                 key_scheme,
                 derivation_path,
             } => {
-                // check if input is a private key -- should start with 0x
-                if input_string.starts_with("0x") {
-                    let bytes: Vec<u8> = Hex::decode(&input_string).map_err(|_| {
-                        anyhow!("Private key is malformed. Importing private key failed.")
-                    })?;
-                    let skp = match key_scheme {
-                        SignatureScheme::ED25519 => {
-                            let kp = Ed25519KeyPair::from_bytes(&bytes)?;
-                            SuiKeyPair::Ed25519(kp)
-                        }
-                        SignatureScheme::Secp256k1 => {
-                            let kp = Secp256k1KeyPair::from_bytes(&bytes)?;
-                            SuiKeyPair::Secp256k1(kp)
-                        }
-                        SignatureScheme::Secp256r1 => {
-                            let kp = Secp256r1KeyPair::from_bytes(&bytes)?;
-                            SuiKeyPair::Secp256r1(kp)
-                        }
-                        _ => return Err(anyhow!("Unsupported scheme")),
-                    };
-                    let key = Key::from(&skp);
-                    keystore.add_key(alias, skp)?;
-                    CommandOutput::Import(key)
-                } else {
-                    let sui_address = keystore.import_from_mnemonic(
-                        &input_string,
-                        key_scheme,
-                        derivation_path,
-                    )?;
-                    let skp = keystore.get_key(&sui_address)?;
-                    let key = Key::from(skp);
-                    CommandOutput::Import(key)
+                if Hex::decode(&input_string).is_ok() {
+                    return Err(anyhow!(
+                        "sui.keystore and Sui Wallet no longer support importing 
+                    private key as Hex, if you are sure your private key is encoded in Hex, use 
+                    `sui keytool convert $HEX` to convert first then import the bech32 encoded 
+                    private key starting with `suiprivkey`."
+                    ));
                 }
-            }
 
+                match SuiKeyPair::decode(&input_string) {
+                    Ok(skp) => {
+                        println!("Importing bech32 encoded private key to keystore");
```

Comment:
Do we really need this `println!`?

Reply:
imo its useful since it interprets the private key in different ways.